server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/SearchUsersAction.java

```diff
@@ -0,0 +1,76 @@
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.qualityprofile.ws;
+
+import java.util.Arrays;
+import org.apache.commons.io.IOUtils;
+import org.sonar.api.resources.Language;
+import org.sonar.api.resources.Languages;
+import org.sonar.api.server.ws.Request;
+import org.sonar.api.server.ws.Response;
+import org.sonar.api.server.ws.WebService;
+
+import static org.sonar.core.util.stream.MoreCollectors.toSet;
+import static org.sonar.server.qualityprofile.ws.QProfileWsSupport.createOrganizationParam;
+import static org.sonarqube.ws.client.qualityprofile.QualityProfileWsParameters.ACTION_SEARCH_USERS;
+import static org.sonarqube.ws.client.qualityprofile.QualityProfileWsParameters.PARAM_LANGUAGE;
+import static org.sonarqube.ws.client.qualityprofile.QualityProfileWsParameters.PARAM_QUALITY_PROFILE;
+
+public class SearchUsersAction implements QProfileWsAction {
+
+  private final Languages languages;
+
+  public SearchUsersAction(Languages languages) {
+    this.languages = languages;
+  }
+
+  @Override
+  public void define(WebService.NewController context) {
+    WebService.NewAction action = context
+      .createAction(ACTION_SEARCH_USERS)
+      .setDescription("List the users that are allowed to edit a Quality Profile.<br>" +
+        "Requires the 'Administer Quality Profiles' permission or the ability to edit the quality profile.")
+      .setHandler(this)
+      .setInternal(true)
+      .addSearchQuery("freddy", "names", "logins")
+      .addSelectionModeParam()
+      .addPagingParams(25)
+      .setResponseExample(getClass().getResource("search_users-example.json"))
+      .setSince("6.6");
+
+    action.createParam(PARAM_QUALITY_PROFILE)
+      .setDescription("Quality Profile name")
+      .setRequired(true)
+      .setExampleValue("Recommended quality profile");
+
+    action
+      .createParam(PARAM_LANGUAGE)
+      .setDescription("Quality profile language")
+      .setRequired(true)
+      .setPossibleValues(Arrays.stream(languages.all()).map(Language::getKey).collect(toSet()));
+
+    createOrganizationParam(action);
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    IOUtils.write(IOUtils.toString(getClass().getResource("search_users-example.json")), response.stream().output());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "write" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1943)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "toString" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1943)